src/Microsoft.DotNet.CodeFormatting/Rules/MarkReadonlyFieldsRule.cs

```diff
@@ -0,0 +1,527 @@
+                        return IsSymbolVisibleOutsideSolution(
+                            symbol.ContainingType,
+                            internalsVisibleToAttribute,
+                            solution);
+                    }
+
+                    // They are public, we are going to skip them.
+                    return true;
+                }
+
+                if (accessibility > Accessibility.Private)
+                {
+                    bool visibleOutsideSolution = IsVisibleOutsideSolution(
+                        symbol,
+                        internalsVisibleToAttribute,
+                        solution);
+
+                    if (visibleOutsideSolution)
+                    {
+                        if (symbol.ContainingType != null)
+                        {
+                            // a visible symbol in a non-visible class isn't visible
+                            return IsSymbolVisibleOutsideSolution(
+                                symbol.ContainingType,
+                                internalsVisibleToAttribute,
+                                solution);
+                        }
+
+                        return true;
+                    }
+                }
+
+                return false;
+            }
+
+            private bool IsVisibleOutsideSolution(
+                ISymbol field,
+                ISymbol internalsVisibleToAttribute,
+                Solution solution)
+            {
+                bool isVisible;
+                IAssemblySymbol assembly = field.ContainingAssembly;
+                if (_visibleOutsideAsembly.TryGetValue(assembly, out isVisible))
+                {
+                    return isVisible;
+                }
+
+                foreach (AttributeData internalsVisibleInstance in assembly.GetAttributes()
+                    .Where(a => Equals(a.AttributeClass, internalsVisibleToAttribute)))
+                {
+                    if (internalsVisibleInstance.ConstructorArguments.Length != 1)
+                    {
+                        // Unexpected number of agruments, isn't really the correct type.
+                        continue;
+                    }
+
+                    TypedConstant assemblyNameArgument = internalsVisibleInstance.ConstructorArguments[0];
+                    if (assemblyNameArgument.Kind != TypedConstantKind.Primitive)
+                    {
+                        // The first argument wasn't a primitave value, isn't really the correct type
+                        continue;
+                    }
+
+                    string assemblyName = assemblyNameArgument.Value as string;
+                    if (String.IsNullOrEmpty(assemblyName))
+                    {
+                        // The first argument wasn't a string, isn't really the correct type
+                        continue;
+                    }
+
+                    if (!solution.Projects.Any(p => ProjectHasName(p, assemblyName)))
```

Comment:
I don't think we can make this check safely.  It would cause different behavior for the formatter when it was run on a project vs. a solution.  Generally the two should be equal.